src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -1464,7 +1464,7 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 		}
 		if (originalIcon !== this.shellLaunchConfig.icon || this.shellLaunchConfig.color) {
```

Comment:
won't the first condition here always be false?

`originalIcon !== this.shellLaunchConfig.icon` 

if i'm missing something and that could be true, could use `this._icon` instead to also capture `this._shellLaunchConfig.attachPersistentProcess`

Reply:
It's not always false, I'm not sure the exact case where this happens but there's some complexity around deferred icon resolving/updating in the terminal tab.